### DISCUSSION TITLE
How to change proration method for a paddle subscription?

### DISCUSSION THREAD

**Discussion by danielschnee** [pay-rails/pay#1114] [Q&A]
Hey. As the title asks: How to change proration method for a paddle subscription?

I read through the source code and found: https://github.com/pay-rails/pay/blob/937b85b55e2c94ae2194dfc28ac03c56a7d05e7d/app/models/pay/paddle_billing/subscription.rb#L162C1-L176C10

So i tried doing the following: `@user.payment_processor.subscription.swap("yearly", proration_billing_mode: "full_next_billing_period")`

But i got an error: "Error 400: Your request was malformed. 'subscription_new_items_not_valid'"

How can i change the proration method? Any suggestions?

**Comment by danielschnee** [comment]
Any chance to get an answer/help to this?

**Reply by excid3** [reply]
Check the Paddle docs?

**Reply by deanpcmad** [reply]
It appears you can do it by just sending an API request - https://developer.paddle.com/build/subscriptions/replace-products-prices-upgrade-downgrade#change-billing-frequency - so the code you've used *should* work, in theory
